Medical LLM Leaderboard - publication EOW (#1989)
huggingface/blog: leaderboard-medicalllm.md

Over the years, Large Language Models (LLMs) have emerged as a groundbreaking technology with immense potential to revolutionize various aspects of healthcare. These models, such as [GPT-3](https://arxiv.org/abs/2005.14165), [GPT-4](https://arxiv.org/abs/2303.08774) and [Med-PaLM 2](https://arxiv.org/abs/2305.09617) have demonstrated remarkable capabilities in understanding and generating human-like text, making them valuable tools for tackling complex medical tasks and improving patient care. They have notably shown promise in various medical applications, such as medical question-answering (QA), dialogue systems, and text generation. Moreover, with the exponential growth of electronic health records (EHRs), medical literature, and patient-generated data, LLMs could help healthcare professionals extract valuable insights and make informed decisions.

However, despite the immense potential of Large Language Models (LLMs) in healthcare, there are significant and specific challenges that need to be addressed. 

> While model errors may be of minimal consequence in casual conversation or other contexts, wrong explanation and answers in the medical domain can have severe consequences for patient care and outcomes. The accuracy and reliability of information provided by language models can be a matter of life or death, as it could potentially affect healthcare decisions, diagnosis, and treatment plans. 

Author: ```suggestion
When models are used for recreational conversational aspects, errors have little repercussions; this is not the case for uses in the medical domain however, where wrong explanation and answers can have severe consequences for patient care and outcomes. The accuracy and reliability of information provided by language models can be a matter of life or death, as it could potentially affect healthcare decisions, diagnosis, and treatment plans.
```

> For example, when given a medical query, GPT-3 provided a wrong answer (as shown below) which could have dreadful consequences if taken at face value. To give some context, although the explanation correctly stated that tetracycline is contraindicated in pregnancy due to potential harmful effects on the fetus, GPT-3.5 still incorrectly selected it as the answer, but if tetracycline were administered based on this incorrect information, it could lead to bone growth problems in the baby.

Author: ```suggestion
For example, when given a medical query (see below), GPT-3 incorrectly recommended tetracycline for a pregnant patient, despite correctly explaining its contraindication due to potential harm to the fetus. Acting on this incorrect recommendation could lead to bone growth problems in the baby.
```

![Image source : [https://arxiv.org/pdf/2311.05112.pdf](https://arxiv.org/abs/2307.15343)](https://prod-files-secure.s3.us-west-2.amazonaws.com/e749ee15-500e-4660-b028-a1069816cfa3/4ea13c24-88a0-40b8-880f-e7bc9888b997/Screenshot_2024-04-12_at_1.18.05_PM.png)

> To fully utilize the power of LLMs in healthcare, it is crucial to develop and benchmark models **in a setup** specifically designed for the medical domain, taking into account the unique characteristics and requirements of healthcare data and applications. The development of methods to evaluate the Medical-LLM is not just of academic interest but of practical importance, given the real-life risks they pose in the healthcare sector.

Author: ```suggestion
To fully utilize the power of LLMs in healthcare, it is crucial to develop and benchmark models using a setup specifically designed for the medical domain. This setup should take into account the unique characteristics and requirements of healthcare data and applications. The development of methods to evaluate the Medical-LLM is not just of academic interest but of practical importance, given the real-life risks they pose in the healthcare sector.
```

Over the years, Large Language Models (LLMs) have emerged as a groundbreaking technology with immense potential to revolutionize various aspects of healthcare. These models, such as [GPT-3](https://arxiv.org/abs/2005.14165), [GPT-4](https://arxiv.org/abs/2303.08774) and [Med-PaLM 2](https://arxiv.org/abs/2305.09617) have demonstrated remarkable capabilities in understanding and generating human-like text, making them valuable tools for tackling complex medical tasks and improving patient care. They have notably shown promise in various medical applications, such as medical question-answering (QA), dialogue systems, and text generation. Moreover, with the exponential growth of electronic health records (EHRs), medical literature, and patient-generated data, LLMs could help healthcare professionals extract valuable insights and make informed decisions.

However, despite the immense potential of Large Language Models (LLMs) in healthcare, there are significant and specific challenges that need to be addressed. 

> While model errors may be of minimal consequence in casual conversation or other contexts, wrong explanation and answers in the medical domain can have severe consequences for patient care and outcomes. The accuracy and reliability of information provided by language models can be a matter of life or death, as it could potentially affect healthcare decisions, diagnosis, and treatment plans. 

Reviewer: first sentence could be rewritten for clarity

Reviewer 2: Thank you for the review. Could you please confirm if this works?

`However, despite the immense potential of Large Language Models (LLMs) in healthcare, there are significant challenges specific to this domain that need to be addressed.`

Author: I think the sentence to be edited was the first of paragraph at line 20, will change it

> For example, when given a medical query, GPT-3 provided a wrong answer (as shown below) which could have dreadful consequences if taken at face value. To give some context, although the explanation correctly stated that tetracycline is contraindicated in pregnancy due to potential harmful effects on the fetus, GPT-3.5 still incorrectly selected it as the answer, but if tetracycline were administered based on this incorrect information, it could lead to bone growth problems in the baby.

Reviewer: last sentence is a bit long

Reviewer 2: For example, when given a medical query, GPT-3 incorrectly recommended tetracycline for a pregnant patient, despite correctly explaining its contraindication due to potential harm to the fetus. Acting on this incorrect recommendation could lead to bone growth problems in the baby.

![Image source : [https://arxiv.org/pdf/2311.05112.pdf](https://arxiv.org/abs/2307.15343)](https://prod-files-secure.s3.us-west-2.amazonaws.com/e749ee15-500e-4660-b028-a1069816cfa3/4ea13c24-88a0-40b8-880f-e7bc9888b997/Screenshot_2024-04-12_at_1.18.05_PM.png)

> To fully utilize the power of LLMs in healthcare, it is crucial to develop and benchmark models **in a setup** specifically designed for the medical domain, taking into account the unique characteristics and requirements of healthcare data and applications. The development of methods to evaluate the Medical-LLM is not just of academic interest but of practical importance, given the real-life risks they pose in the healthcare sector.

Reviewer: first sentence is a bit hard to read

Reviewer 2: To fully utilize the power of LLMs in healthcare, it is crucial to develop and benchmark models using a setup specifically designed for the medical domain. This setup should take into account the unique characteristics and requirements of healthcare data and applications.

<script type="module" src="https://gradio.s3-us-west-2.amazonaws.com/4.20.1/gradio.js"> </script>
<gradio-app theme_mode="light" space="openlifescienceai/open_medical_llm_leaderboard"></gradio-app>

The Open Medical-LLM Leaderboard aims to address these challenges and limitations by providing a standardized platform for evaluating and comparing the performance of various large language models on a diverse range of medical tasks and datasets. By offering a comprehensive assessment of each model's medical knowledge and question-answering capabilities, the leaderboard aims to foster the development of more effective and reliable medical LLMs. 

This platform enables researchers and practitioners to identify the strengths and weaknesses of different approaches, drive further advancements in the field, and ultimately contribute to better patient care and outcomes

## Datasets, Tasks, and Evaluation Setup 

The Medical-LLM Leaderboard includes a variety of tasks, and uses accuracy as its primary evaluation metric (accuracy measures the percentage of correct answers provided by a language model across the various medical QA datasets).

### MedQA (USMLE)

> The [MedQA](https://arxiv.org/abs/2009.13081) (USMLE) dataset consists of multiple-choice questions from the United States Medical Licensing Examination (USMLE). It covers general medical knowledge and includes 11,450 questions in the development set and 1,273 questions in the test set. Each question has 4 or 5 answer choices, and the dataset is designed to assess the medical knowledge and reasoning skills required for medical licensure in the United States.

Reviewer: ```suggestion
The [MedQA](https://arxiv.org/abs/2009.13081) dataset consists of multiple-choice questions from the United States Medical Licensing Examination (USMLE). It covers general medical knowledge and includes 11,450 questions in the development set and 1,273 questions in the test set. Each question has 4 or 5 answer choices, and the dataset is designed to assess the medical knowledge and reasoning skills required for medical licensure in the United States.
```

![Image source : [https://arxiv.org/pdf/2311.05112.pdf](https://arxiv.org/abs/2402.07023)](https://prod-files-secure.s3.us-west-2.amazonaws.com/e749ee15-500e-4660-b028-a1069816cfa3/174711e7-e27b-431f-88a8-cc3fbcc64c33/a00c20bc-f459-4f5a-b236-afd77f3557c2_.png)

Google's model, [Gemini Pro](https://arxiv.org/abs/2312.11805) demonstrates strong performance in various medical domains, particularly excelling in data-intensive and procedural tasks like Biostatistics, Cell Biology, and Obstetrics & Gynecology. However, it shows moderate to low performance in critical areas such as Anatomy, Cardiology, and Dermatology, revealing gaps that require further refinement for comprehensive medical application.

![Image source : [https://arxiv.org/pdf/2311.05112.pdf](https://arxiv.org/abs/2402.07023)](https://prod-files-secure.s3.us-west-2.amazonaws.com/e749ee15-500e-4660-b028-a1069816cfa3/dbc10ab2-e1b1-4f80-ba8d-25d5a37203e5/Screenshot_2024-04-13_at_12.29.29_AM.png)

## Submitting Your Model for Evaluation

To submit your model for evaluation on the Open Medical-LLM Leaderboard, follow these steps:

> 1. **Convert Model Weights to Safetensors Format**

Reviewer: ```suggestion
**1. Convert Model Weights to Safetensors Format**
```

First, convert your model weights to the safetensors format. Safetensors is a new format for storing weights that is safer and faster to load and use. Converting your model to this format will also allow the leaderboard to display the number of parameters of your model in the main table.

> 2. **Ensure Compatibility with AutoClasses**

Reviewer: ```suggestion
**2. Ensure Compatibility with AutoClasses**
```

Before submitting your model, make sure you can load your model and tokenizer using the AutoClasses from the Transformers library. Use the following code snippet to test the compatibility:

```python
from transformers import AutoConfig, AutoModel, AutoTokenizer

> config = AutoConfig.from_pretrained("your model name")

Reviewer: ```suggestion
config = AutoConfig.from_pretrained(MODEL_HUB_ID)
```

model = AutoModel.from_pretrained("your model name")
tokenizer = AutoTokenizer.from_pretrained("your model name")

```

If this step fails, follow the error messages to debug your model before submitting it. It's likely that your model has been improperly uploaded.

> 3. Make Your Model Public**

Reviewer: ```suggestion
**3. Make Your Model Public**
```

Ensure that your model is publicly accessible. The leaderboard cannot evaluate models that are private or require special access permissions.

> 4. Remote Code Execution (Coming Soon)**

Reviewer: ```suggestion
**4. Remote Code Execution (Coming Soon)**
```

Currently, the Open Medical-LLM Leaderboard does not support models that require `use_remote_code=True`. However, the leaderboard team is actively working on adding this feature, so stay tuned for updates.

> 5. Submit Your Model via the Leaderboard Website**

Reviewer: ```suggestion
**5. Submit Your Model via the Leaderboard Website**
```

![Screenshot 2024-04-13 at 3.02.10 PM.png](https://prod-files-secure.s3.us-west-2.amazonaws.com/e749ee15-500e-4660-b028-a1069816cfa3/7d48f302-53da-40bd-bd9e-23aa8eb83414/Screenshot_2024-04-13_at_3.02.10_PM.png)

## Citation

If you find our evaluations useful, please consider citing our work

**Medical-LLM Leaderboard**
```
@misc{Medical-LLM Leaderboard,
author = {Ankit Pal, Pasquale Minervini and Andreas Geert Motzfeldt},
title = {openlifescienceai/open_medical_llm_leaderboard},
year = {2024},
publisher = {Hugging Face},
howpublished = "\url{https://huggingface.co/spaces/openlifescienceai/open_medical_llm_leaderboard}"
}
```

> **Med-PaLM Paper**

Author: Not sure it makes sense to have this citation since the leaderboard is not related
